### PR TITLE
Implemented UpdateFields task for Jira. Added OAuth authorization

### PR DESCRIPTION
**Pull request by iNikitaGricenko**
### What changes are being made and why?

- Implemented task to update fields in Jira ticket (#10)
- Added OAuth authorization using accessToken
---

### How the changes have been QAed?

```yaml
id: jira_update_field
namespace: company.myteam
tasks:
  - id: update_ticket_field
    type: io.kestra.plugin.jira.issues.UpdateFields
    baseUrl: your-domain.atlassian.net
    accessToken: "{{ secret('my_jira_access_token') }}"
    issueIdOrKey: YOUR_ISSUE_KEY
    fields:
      description: "Updated description of: {{ execution.id }}"
      customfield_10005: "Updated value"
```

---

### CODE REVIEW

File: src/main/java/io/kestra/plugin/jira/issues/JiraClient.java [hunk 1]
```diff
@@ -30,18 +34,25 @@ public abstract class JiraClient extends Task implements RunnableTask<VoidOutput
...
 
     @Schema(
-        title = "Atlassian password"
+        title = "Atlassian password or API token",
+        description = "(Required for basic & API token authorization)"
     )
     @PluginProperty(dynamic = true)
-    @NotBlank
-    protected String password;
+    protected String passwordOrToken;
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/jira/issues/JiraClient.java:48]: Thsi plugin has already been published in 0.18 so you cannot change a property name for backward compatibility

File: src/main/java/io/kestra/plugin/jira/issues/UpdateFields.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import java.util.Map;
+import java.util.Objects;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode(callSuper = true)
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Update a Jira field",
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/jira/issues/UpdateFields.java:31]:
```suggestion
    title = "Update Jira fields",
```

File: src/main/java/io/kestra/plugin/jira/issues/UpdateFields.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    description = "Updates a specific field in a Jira ticket."
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/jira/issues/UpdateFields.java:32]:
```suggestion
    description = "Update specific fields in a Jira ticket."
```

File: src/main/java/io/kestra/plugin/jira/issues/UpdateFields.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Update a Jira ticket field",
```
Comment (loicmathieu) [inline src/main/java/io/kestra/plugin/jira/issues/UpdateFields.java:37]:
```suggestion
            title = "Update a Jira ticket fields",
```

**Review by loicmathieu — CHANGES_REQUESTED** [review]

**Comment by anna-geller** [comment]
closes https://github.com/kestra-io/plugin-jira/issues/10

**Review by loicmathieu — APPROVED** [review]
LGTM

**Pull request merged**